sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/GuidUtilities.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace Azure.Messaging.ServiceBus.Primitives;
+
+internal static class GuidUtilities
```

Comment:
Should we consider moving this to Core?   May be situational, but definitely general purpose.

Reply:
There are some nuances that are somewhat idiomatic to the SB use case, e.g. returning default rather than throwing when trying to parse a Guid from a buffer that is not 16 bytes. That said, we can probably rethink whether we need this behavior. I will look into moving this to Core.

Reply:
My implementation is more aligned with the runtime code so be aware.

Reply:
Yep, I noticed that. I think it is probably okay as receiving a message with a malformed delivery tag would make it unusable in most scenarios anyway.